drivers/azure/azureutil/azureutil.go

```diff
@@ -460,6 +460,21 @@ func (a AzureClient) removeOSDiskBlob(resourceGroup, vmName, vhdURL string) erro
 	if err != nil {
 		log.Debugf("Container remove happened: %v", ok)
 	}
+
+	cts, err := bs.GetBlobService().ListContainers(blobstorage.ListContainersParameters{})
+	if err != nil {
+		log.Debugf("Error listing containers.")
+	}
+
+	if len(cts.Containers) == 0 {
+		resp, err := a.storageAccountsClient().Delete(resourceGroup, storageAccount)
+		if err != nil {
+			return err
+		}
+
+		log.Debugf("No storage containers left. Virtual machine storage account deletion happened: %v", resp.Response.Status)
```

Comment:
Move this before the delete call, and change the message to something like `No storage containers left. Deleting virtual machine storage account`

We can then add another debug message that reports the response status code.

Reply:
Yes, that is correct, I will revise it! Thank you for your help!

Reply:
@-shin What is your opinion?

Reply:
@shin- what is your opinion?